src/backend/distributed/commands/table.c

```diff
@@ -2108,6 +2129,70 @@ PostprocessAlterTableStmt(AlterTableStmt *alterTableStatement)
 }
 
 
+/*
+ * PostprocessAlterTableStmtIndexNames runs after the ALTER TABLE command
+ * has already run on the coordinator, and also after the distributed DDL
+ * Jobs have been executed on the workers.
+ *
+ * We might have wrong index names generated on indexes of shards of partitions,
+ * see https://github.com/citusdata/citus/pull/5397 for the details. So we
+ * perform the relevant checks and index renaming here.
+ */
+void
+PostprocessAlterTableStmtIndexNames(AlterTableStmt *alterTableStatement)
```

Comment:
Maybe Postprocess -> Fix?

Given that it already has IndexNames in the name, this function is not meant to be a generic post-processor.